### PR TITLE
Update use of self in best practices.

### PR DESCRIPTION
**Pull request by roberthein**
Not writing `self` explicitly is the way to go from now on.

### CODE REVIEW

**Comment by elland** [comment]
We need better language and a more sound reasoning. 👌🏽

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -550,7 +550,7 @@ Additional capabilities of classes:
 
 ### Use of Self
 
-Always use `self` when referencing properties. It will make your life simpler. Trust us.
+We don’t write `self` explicitly, because Swift assumes that you are referring to a property or method of the current instance whenever you use a known property or method name within a method.
```
Comment (marijnschilling) [inline BEST_PRACTICES.md]: Change this to `within a class` maybe? 

And I'd like to add (something like) this:
for the team it was a choice between: 
- not using `self` for the reason that it removes unnecessary verboseness and is the swift guidelines from apple 
- use `self` for code clarity reasons.

**Comment by roberthein** [comment]
@marijnschilling en @Elland I updated the reasoning. If you have any suggestions let me know.

**Comment by elland** [comment]
https://twitter.com/arekholko/status/883055999023763456

**Pull request closed without merging**